controller/volunteer_controller.py

```diff
@@ -27,25 +41,74 @@ def view_volunteer_profile(volunteer: Volunteer) -> str:
     pass
 
 
-def edit_volunteer_profile(volunteer: Volunteer, firstname: str, lastname: str,
-                           phone: str) -> Volunteer:
-    """
-    Yunsy, Yingbo
-    A function used by admin and volunteer
-    Should include re-assign volunteer to a camp
-    """
-    pass
-
-
-def change_volunteer_camp(volunteer: Volunteer, camp: Camp, is_admin: bool) -> Volunteer:
-    """
-    The admin role
-    :param is_admin: if this method is called by an admin
-    :param volunteer:
-    :param camp: the new camp
-    :return:
-    """
-    pass
+def edit_firstname(volunteer: Volunteer, firstname: str) -> Volunteer:
+    if len(firstname) > 1:
+        volunteer_called = volunteer
+        volunteer_called.firstname = firstname
+        volunteer_called.save()
+        return volunteer_called
+    else:
+        raise ControllerError(f"Invalid name: {firstname}. First name should be more than 1 character.")
+
+
+def edit_lastname(volunteer: Volunteer, lastname: str) -> Volunteer:
+    if len(lastname) > 1:
+        volunteer_called = volunteer
+        volunteer_called.lastname = lastname
+        volunteer_called.save()
+        return volunteer_called
+    else:
+        raise ControllerError(f"Invalid name: {lastname}. Last name should be more than 1 character.")
+
+
+def edit_phone(volunteer: Volunteer, phone: str) -> Volunteer:
+    phone_text = r"\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|2[98654321]\d|9[8543210]|8[6421]" \
+                 r"|6[6543210]|5[87654321]|4[987654310]|3[9643210]|2[70]|7|1)\d{5,14}$"
+    if re.search(phone_text, phone):
+        volunteer_called = volunteer
+        volunteer_called.phone = phone
+        volunteer_called.save()
+        return volunteer_called
+    else:
+        raise ControllerError(
+            f"Invalid phone number: {phone}. Phone number should start with a plus sign and international code")
```

Comment:
Quick comment:
It's best to not repeat validation code from the Volunteer class in here. 
Solutions are:
1. make validation methods in Volunteer public (and classmethod), and call them here
2. Change the Volunteer class so that all assignment to those variables must first pass through validations. To do that, you may override [\_\_setattr\_\_](https://python-reference.readthedocs.io/en/latest/docs/dunderattr/setattr.html)

Reply:
Thanks for the suggestion! I made the validation methods in Volunteer public and class methods to minimize code repetition.